File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SpawnExperienceOrbPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record SpawnExperienceOrbPacket(int entityId, int x, int y, int z, short count)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeVarInt(os, entityId);
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
        os.writeShort(count);
    }

    @Override
    public byte getId() {
        return 0x11;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SpawnExperienceOrbPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SpawnExperienceOrbPacket(val entityId: Int, val x: Int, val y: Int, val z: Int, val count: Short) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        writeVarInt(os, entityId)
        os.writeInt(x)
        os.writeInt(y)
        os.writeInt(z)
        os.writeShort(count.toInt())
    }

    override fun getId(): Byte {
        return 0x11
    }
}
